src/npe2/manifest/_bases.py

```diff
@@ -74,13 +75,18 @@ def from_file(cls, path: Union[Path, str]):
         if path.suffix.lower() == ".json":
             loader = json.load
         elif path.suffix.lower() == ".toml":
-            loader = toml.load
+            try:
+                import tomllib
+            except ImportError:
+                import tomli as tomllib  # type: ignore [no-redef]
+
+            loader = tomllib.load
         elif path.suffix.lower() in (".yaml", ".yml"):
             loader = yaml.safe_load
         else:
             raise ValueError(f"unrecognized file extension: {path}")  # pragma: no cover
 
-        with open(path, encoding="utf-8") as f:
+        with open(path, mode="rb") as f:
```

Comment:
why change form text to binary?

Reply:
because `.load` requires it, and otherwise you get an exception

Reply:
And `yaml.safe_loader` accept both binary and text format? I try to check the documentation but not found. Only `stream` argument without explanation.